config-service-impl/src/main/java/org/hypertrace/config/service/store/DocumentConfigStore.java

```diff
@@ -245,7 +245,7 @@ public long getMatchingConfigsCount(
     return collection.count(query);
   }
 
-  private Query buildQuery(
+  Query buildQuery(
```

Comment:
nit - don't expose a method just to test it. Test it through the public interface.

Reply:
It's bit involved to write cases with public interface, can take it up as next PR.

Reply:
Next PR is fine.

As far as what to change, one option would be to extract a query builder class, in which case testing that makes sense in a unit test (while tests on the store would start mocking it). But an even easier option... this test could just call `getAllConfigs` instead - it passes the same 4 arguments through to the query builder directly. Then just verify the call to `collection.query`.

Reply:
I know this, since i need to wait for one more day to get it merged, i thought of doing in next PR. Anyway i have addressed all the comments. We should also take decision to remove sorting on ConfigVersion, i don't its needed any more.